src/start/IosCompliantAppleAuthButton/index.js

```diff
@@ -45,40 +32,35 @@ type State = $ReadOnly<{|
  * Apple", but without marking it with a different style from the
  * other buttons.
  */
-class IosCompliantAppleAuthButton extends PureComponent<Props, State> {
-  state = {
-    isNativeButtonAvailable: undefined,
-  };
+export default function IosCompliantAppleAuthButton(props: Props) {
+  const { style, onPress } = props;
+  const theme = useSelector(state => getSettings(state).theme);
+  const [isNativeButtonAvailable, setIsNativeButtonAvailable] = useState<boolean | void>(undefined);
 
-  async componentDidMount() {
-    this.setState({ isNativeButtonAvailable: await AppleAuthentication.isAvailableAsync() });
-  }
-
-  render() {
-    const { style, onPress, theme } = this.props;
-    const { isNativeButtonAvailable } = this.state;
-    if (isNativeButtonAvailable === undefined) {
-      return <View style={[{ height: 44 }, style]} />;
-    } else if (isNativeButtonAvailable) {
-      return (
-        <AppleAuthentication.AppleAuthenticationButton
-          buttonType={AppleAuthentication.AppleAuthenticationButtonType.SIGN_IN}
-          buttonStyle={
-            theme === 'default'
-              ? AppleAuthentication.AppleAuthenticationButtonStyle.WHITE_OUTLINE
-              : AppleAuthentication.AppleAuthenticationButtonStyle.BLACK
-          }
-          cornerRadius={22}
-          style={[{ height: 44 }, style]}
-          onPress={onPress}
-        />
-      );
-    } else {
-      return <Custom style={style} onPress={onPress} theme={theme} />;
+  useEffect(() => {
+    async function getAndSetIsAvailable() {
+      setIsNativeButtonAvailable(await AppleAuthentication.isAvailableAsync());
```

Comment:
Hmm, this nesting is kind of odd-looking.

Google tells me that if you just make the `useEffect` callback an async function, you get a warning:
https://stackoverflow.com/questions/53332321/react-hook-warnings-for-async-function-in-useeffect-useeffect-function-must-ret
so I guess that's why you didn't do that. And this thread is informative:
https://github.com/facebook/react/issues/14326

In general this can be a bad pattern, but it's fine here because this always asks the same question, and the answer can't change in the lifetime of the component -- we can't end up with data that's mismatched or out of sync with the rest of the component.

I guess if/when React delivers "suspense", this will be a spot that can switch to that.

For now: maybe just a link to that React issue thread, to explain this odd-looking pattern?